ambry-store/src/main/java/com.github.ambry.store/StoreMetrics.java

```diff
@@ -143,7 +143,13 @@ public Long getValue() {
         return index.hardDeleteCaughtUp() ? 1L : 0L;
       }
     };
-    registry.register(MetricRegistry.name(PersistentIndex.class, name + "-hardDeleteCaughtUp"),
-        hardDeleteCaughtUp);
+    registry.register(MetricRegistry.name(PersistentIndex.class, name + ".HardDeleteCaughtUp"), hardDeleteCaughtUp);
+  }
+
+  static String getPartitionName(String replicaDir) {
```

Comment:
This is used to extract the last part of a replica path, which is the partition name. E.g.,
`getPartitionName("/mnt/AmbryData/1520")` will return `1520`. `1520` is the partition name.